app/utilities/metadata_parser.py

```diff
@@ -146,6 +146,27 @@ def convert_schema_name(self, data, **kwargs):
             )
         return data
 
+    @post_load
+    def metadata_response_id(
+        self, data, **kwargs
+    ):  # pylint: disable=no-self-use, unused-argument
+        """
+        If response_id is not present : Build response_id from ru_ref,collection_exercise_sid,eq_id and form_type
```

Comment:
Would swap these over for code flow as the first return is response_id is present